### PR TITLE
Fix compilation for wasm32-wasi

### PR DESCRIPTION
**Pull request by TerrorJack**
This one is a rework of #240 and shouldn't break CI in this repo or any other GHC platforms now.

### CODE REVIEW

**Comment by TerrorJack** [comment]
@snoyberg Would you allow the workflow to run on this PR? Thanks :)

**Comment by snoyberg** [comment]
PR is passing CI, and I don't see any issues with this. @bgamari would you mind confirming that you're OK with this change landing?

**Comment by TerrorJack** [comment]
Ben is on holiday yet; also allow me to draft it atm, there were spurious "redundant import" warnings that turned to errors when testing in non-wasm jobs, so I also need a bit of time to look into it. Will undraft and ping again once I'm absolutely sure it doesn't break stuff on other platforms :/

**Comment by TerrorJack** [comment]
Ping; the current patch is good to go :)

File: System/Process.hs [hunk 1]
```diff
@@ -857,6 +862,19 @@ terminateProcess ph = do
 -- ----------------------------------------------------------------------------
 -- Interface to C bits
 
+#if defined(__wasm__)
+
+c_terminateProcess :: PHANDLE -> IO CInt
+c_terminateProcess _ = ioError (ioeSetLocation unsupportedOperation "terminateProcess")
```
Comment (bgamari) [inline System/Process.hs:868]: Yes, this is better than the previous iteration. I probably would have factored out this pattern into a `unsupportedByWasm` helper function living in `System.Process.Internal` but I'm fine with this as well.

**Review by bgamari — APPROVED** [review]

**Pull request merged**